rest/method_item_put_test.go

```diff
@@ -294,6 +363,179 @@ func TestPutItem(t *testing.T) {
+			ExtraTest:    checkPayload("foo", "1", map[string]interface{}{"id": "1", "foo": "baz", "bar": "value"}),
+		},
+		`pathID:found,body:valid,required:missing`: {
+			Init: initForRequiredField(now, yesterday),
+			NewRequest: func() (*http.Request, error) {
+				body := bytes.NewReader([]byte(`{"foo": "baz"}`))
+				return http.NewRequest("PUT", "/foo/1", body)
+			},
+			ResponseCode: http.StatusUnprocessableEntity,
+			ResponseBody: `{
+				"code": 422,
+				"message": "Document contains error(s)",
+				"issues": {
+					"bar": ["required"]
+				}
+			}`,
+		},
+		`pathID:found,body:valid,required:change`: {
+			Init: initForRequiredField(now, yesterday),
+			NewRequest: func() (*http.Request, error) {
+				body := bytes.NewReader([]byte(`{"foo": "baz", "bar": "value1"}`))
+				return http.NewRequest("PUT", "/foo/2", body)
+			},
+			ResponseCode: http.StatusOK,
+			ResponseBody: `{"id": "2", "foo": "baz", "bar": "value1"}`,
+			ExtraTest:    checkPayload("foo", "2", map[string]interface{}{"id": "2", "foo": "baz", "bar": "value1"}),
+		},
+		`pathID:found,body:valid,required:delete`: {
+			Init: initForRequiredField(now, yesterday),
+			NewRequest: func() (*http.Request, error) {
+				body := bytes.NewReader([]byte(`{"foo": "baz"}`))
+				return http.NewRequest("PUT", "/foo/2", body)
+			},
+			ResponseCode: http.StatusUnprocessableEntity,
+			ResponseBody: `{
+				"code": 422,
+				"message": "Document contains error(s)",
+				"issues": {
+					"bar": ["required"]
+				}
+			}`,
+		},
+
+		`pathID:not-found,body:valid,required-default:missing`: {
+			Init: initForRequiredDefaultField(now, yesterday),
+			NewRequest: func() (*http.Request, error) {
+				body := bytes.NewReader([]byte(`{"foo": "baz"}`))
+				return http.NewRequest("PUT", "/foo/66", body)
+			},
+			ResponseCode: http.StatusCreated,
+			ResponseBody: `{"id": "66", "foo": "baz", "bar": "default"}`,
+			ExtraTest:    checkPayload("foo", "66", map[string]interface{}{"id": "66", "foo": "baz", "bar": "default"}),
+		},
+		`pathID:not-found,body:valid,required-default:set`: {
+			Init: initForRequiredDefaultField(now, yesterday),
+			NewRequest: func() (*http.Request, error) {
+				body := bytes.NewReader([]byte(`{"foo": "baz", "bar": "value"}`))
+				return http.NewRequest("PUT", "/foo/1", body)
+			},
+			ResponseCode: http.StatusOK,
+			ResponseBody: `{"id": "1", "foo": "baz", "bar": "value"}`,
+			ExtraTest:    checkPayload("foo", "1", map[string]interface{}{"id": "1", "foo": "baz", "bar": "value"}),
+		},
+		`pathID:found,body:valid,required-default:missing`: {
+			Init: initForRequiredDefaultField(now, yesterday),
+			NewRequest: func() (*http.Request, error) {
+				body := bytes.NewReader([]byte(`{"foo": "baz"}`))
+				return http.NewRequest("PUT", "/foo/1", body)
+			},
+			ResponseCode: http.StatusUnprocessableEntity,
+			ResponseBody: `{
+				"code": 422,
+				"message": "Document contains error(s)",
+				"issues": {
+					"bar": ["required"]
```

Comment:
Same here, I think the `"bar"` should be set to `"default"` here, so that if I repeat a PUT request twise, it's semantically the same (or very similar) to running it once.

However, if this proves hard to achieve, this functionality is a lot better than silently clearing the field and is preferable over the current way in which things works.

Reply:
Actually this is kinda impossible to happen.  For (Required,Default) we can't have existing document without that field set ( ID:"1"). I did this test for completeness sake.